trunk/src/app/srs_app_rtc_source.cpp

```diff
@@ -1481,38 +1505,42 @@ SrsRtcFrameBuilder::SrsRtcFrameBuilder(ISrsStreamBridge* bridge)
 {
     bridge_ = bridge;
     is_first_audio_ = true;
-    codec_ = NULL;
+    audio_transcoder_ = NULL;
+    video_codec_ = SrsVideoCodecIdAVC;
     header_sn_ = 0;
     memset(cache_video_pkts_, 0, sizeof(cache_video_pkts_));
     rtp_key_frame_ts_ = -1;
     sync_state_ = -1;
-    obs_whip_sps_ = obs_whip_pps_ = NULL;
+    obs_whip_vps_ = obs_whip_sps_ = obs_whip_pps_ = NULL;
 }
 
 SrsRtcFrameBuilder::~SrsRtcFrameBuilder()
 {
-    srs_freep(codec_);
+    srs_freep(audio_transcoder_);
     clear_cached_video();
+    srs_freep(obs_whip_vps_);
     srs_freep(obs_whip_sps_);
     srs_freep(obs_whip_pps_);
 }
 
-srs_error_t SrsRtcFrameBuilder::initialize(SrsRequest* r)
+srs_error_t SrsRtcFrameBuilder::initialize(SrsRequest* r, SrsAudioCodecId audio_codec, SrsVideoCodecId video_codec)
 {
     srs_error_t err = srs_success;
 
-    srs_freep(codec_);
-    codec_ = new SrsAudioTranscoder();
+    srs_freep(audio_transcoder_);
+    audio_transcoder_ = new SrsAudioTranscoder();
 
     SrsAudioCodecId from = SrsAudioCodecIdOpus; // TODO: From SDP?
     SrsAudioCodecId to = SrsAudioCodecIdAAC; // The output audio codec.
     int channels = 2; // The output audio channels.
     int sample_rate = 48000; // The output audio sample rate in HZ.
     int bitrate = _srs_config->get_rtc_aac_bitrate(r->vhost); // The output audio bitrate in bps.
-    if ((err = codec_->initialize(from, to, channels, sample_rate, bitrate)) != srs_success) {
+    if ((err = audio_transcoder_->initialize(audio_codec, to, channels, sample_rate, bitrate)) != srs_success) {
```

Comment:
```suggestion
    SrsAudioCodecId from = audio_codec;
    if ((err = audio_transcoder_->initialize(from, to, channels, sample_rate, bitrate)) != srs_success) {
```
The srs only support `opus` as audio codec, but I also suggest add a audio codec check, make sure `from != to` or from = AAC.

Reply:
You are right, but it is not imperative for now. Let's revisit this when I come to implement RTMP with OPUS payload in the future.

Reply:
You are right, but it is not necessary at the moment. When I eventually implement RTMP with OPUS payload support, I will comprehensively refine this logic.

Reply:
It is advisable to include a comment line here for future reference, as it may be forgotten otherwise.

`TRANS_BY_GPT4`

Reply:
ok